stratum/lib/security/certificate.cc

```diff
@@ -9,14 +9,16 @@
 #include "openssl/evp.h"
 #include "openssl/pem.h"
 #include "openssl/rsa.h"
+#include "openssl/x509.h"
+#include "openssl/x509v3.h"
 #include "stratum/lib/macros.h"
 
 namespace stratum {
 
-using BIGNUM_ptr = std::unique_ptr<BIGNUM, decltype(&::BN_free)>;
-using BIO_ptr = std::unique_ptr<BIO, decltype(&::BIO_free)>;
-using EVP_PKEY_ptr = std::unique_ptr<EVP_PKEY, decltype(&::EVP_PKEY_free)>;
-using X509_ptr = std::unique_ptr<X509, decltype(&::X509_free)>;
+/* TODO(bocon): Copied from a newer version of x509.h
+                Remove when present in the version we use.
+ */
+#define X509_VERSION_3 2
```

Comment:
Can you explain how this define works? Normally does go before the header include, no?

Reply:
It's just a define that can be used to set the version number.

```cpp
  CHECK_RETURN_IF_FALSE(X509_set_version(unsigned_cert, X509_VERSION_3));
```

https://github.com/openssl/openssl/blob/79b2a2f2eedb9d6b24a3f6748332328cf54568fb/include/openssl/x509.h.in#L636

Reply:
Got it. I was on a completely different trail, like `_GNU_SOURCE` or similar.

Reply:
I guess we'd one of those commits:
https://github.com/google/boringssl/commit/89d1241487fb6b715f6516a1f107e846dd760a85
https://github.com/google/boringssl/commit/a96f4dd3822efdfbd371abc4d8ad0e1b9b497866#diff-878093ea6426091505b4c49c59b78924f42859af0eb4ce39b8089bda9577e013R456-R460

Reply:
I guess we'd one of those commits:
https://github.com/google/boringssl/commit/89d1241487fb6b715f6516a1f107e846dd760a85
https://github.com/google/boringssl/commit/a96f4dd3822efdfbd371abc4d8ad0e1b9b497866#diff-878093ea6426091505b4c49c59b78924f42859af0eb4ce39b8089bda9577e013R456-R460